service/src/integrationTest/kotlin/fi/espoo/evaka/ChildBuilder.kt

```diff
@@ -0,0 +1,280 @@
+        private val today: LocalDate,
+        private val childFixture: ChildFixture
+    ) {
+        private var employeeId: UUID? = null
+        private var factor: Double = 1.0
+        private var from: Int = 0
+        private var to: Int = 0
+
+        fun withFactor(factor: Double) = this.apply { this.factor = factor }
+        fun fromDay(day: Int) = this.apply { this.from = day }
+        fun toDay(day: Int) = this.apply { this.to = day }
+        fun createdBy(employeeId: UUID) = this.apply { this.employeeId = employeeId }
+
+        fun execAndReturn(): ChildFixture {
+            tx.insertTestAssistanceNeed(
+                DevAssistanceNeed(
+                    childId = childFixture.childId,
+                    updatedBy = employeeId ?: throw IllegalStateException("createdBy not set"),
+                    startDate = today.plusDays(from.toLong()),
+                    endDate = today.plusDays(to.toLong()),
+                    capacityFactor = factor
+                )
+            )
+            return childFixture
+        }
+    }
+
+    class AbsenceBuilder(
+        private val tx: Database.Transaction,
+        private val today: LocalDate,
+        private val childFixture: ChildFixture
+    ) {
+        private var day: Int = 0
+        private var type: AbsenceType? = null
+        private var careTypes: List<CareType>? = null
+
+        fun onDay(day: Int) = this.apply { this.day = day }
+        fun ofType(type: AbsenceType) = this.apply { this.type = type }
+        fun forCareTypes(vararg careTypes: CareType) = this.apply { this.careTypes = careTypes.asList() }
+
+        private fun exec() {
+            careTypes?.forEach { careType ->
+                tx.insertTestAbsence(
+                    childId = childFixture.childId,
+                    date = today.plusDays(day.toLong()),
+                    absenceType = type ?: throw IllegalStateException("absence type not set"),
+                    careType = careType
+                )
+            } ?: throw IllegalStateException("care types not set")
+        }
+
+        fun execAndReturn(): ChildFixture {
+            exec()
+            return childFixture
+        }
+
+        fun andAnother(): AbsenceBuilder {
+            exec()
+            return AbsenceBuilder(tx, today, childFixture)
+        }
+    }
+
+    class PlacementPlanBuilder(
+        private val tx: Database.Transaction,
+        private val today: LocalDate,
+        private val childFixture: ChildFixture
+    ) {
+        private var unitId: UUID? = null
+        private var type: PlacementType? = null
+        private var from: Int = 0
+        private var to: Int = 0
+        private var deleted = false
+
+        fun toUnit(id: UUID) = this.apply { this.unitId = id }
+        fun ofType(type: PlacementType) = this.apply { this.type = type }
+        fun fromDay(day: Int) = this.apply { this.from = day }
+        fun toDay(day: Int) = this.apply { this.to = day }
```

Comment:
I don't think this is a very obvious API.

Reply:
Maybe change to LocalDate and add these relative ones as overloads? Renaming to fromRelativeDay would be more explicit but the line lengths would become a problem.